models/time_cache.cr

```diff
@@ -0,0 +1,26 @@
+struct TimeCache(K, V)
```

Comment:
Makes sense. My only remaining concern with this is that it makes a request per sorting type, instead of only one and doing the sorting on our side.

But thinking it through, I guess it's not a big problem, since we only make the request for the first request in every half an hour. It should be good enough for now.